beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -243,6 +248,16 @@ func (vs *Server) getPayloadHeaderFromBuilder(ctx context.Context, slot primitiv
 		return nil, fmt.Errorf("incorrect parent hash %#x != %#x", header.ParentHash(), h.BlockHash())
 	}
 
+	reg, err := vs.BlockBuilder.RegistrationByValidatorID(ctx, idx)
+	if err != nil {
+		log.WithError(err).Warn("Proposer: failed to get registration by validator ID, could not check gas limit")
+	} else {
+		gasLimit := expectedGasLimit(parentGasLimit, reg.GasLimit)
+		if gasLimit != header.GasLimit() {
+			return nil, fmt.Errorf("incorrect header gas limit %d != %d", gasLimit, header.GasLimit())
+		}
+	}
```

Comment:
You don't seem to need this else either:
```suggestion
	reg, err := vs.BlockBuilder.RegistrationByValidatorID(ctx, idx)
	if err != nil {
		log.WithError(err).Warn("Proposer: failed to get registration by validator ID, could not check gas limit")
	}
	gasLimit := expectedGasLimit(parentGasLimit, reg.GasLimit)
	if gasLimit != header.GasLimit() {
		return nil, fmt.Errorf("incorrect header gas limit %d != %d", gasLimit, header.GasLimit())
	}
```

Reply:
I think you need the else, we dont want to call `expectedGasLimit` if error

Reply:
reg is used so if there' s an error the call to `expectedGasLimit`  can' t be carried